it/uk/gov/hmrc/transitmovementsrouter/connectors/UpscanConnectorSpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.connectors
+
+import akka.stream.scaladsl.Sink
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import com.github.tomakehurst.wiremock.client.WireMock.get
+import org.scalacheck.Gen
+import org.scalatest.concurrent.PatienceConfiguration
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.time.SpanSugar.convertIntToGrainOfTime
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import play.api.http.Status.NOT_FOUND
+import play.api.http.Status.OK
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.test.HttpClientV2Support
+import uk.gov.hmrc.transitmovementsrouter.it.base.TestActorSystem
+import uk.gov.hmrc.transitmovementsrouter.it.base.WiremockSuite
+import uk.gov.hmrc.transitmovementsrouter.models.responses.UpscanResponse.DownloadUrl
+import uk.gov.hmrc.transitmovementsrouter.services.error.UpscanError
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class UpscanConnectorSpec
+    extends AnyFreeSpec
+    with Matchers
+    with WiremockSuite
+    with TestActorSystem
+    with HttpClientV2Support
+    with ScalaFutures
+    with ScalaCheckDrivenPropertyChecks {
+
+  implicit val timeout: PatienceConfiguration.Timeout = Timeout(3.seconds)
+
+  "streamFile" - {
+
+    implicit val hc: HeaderCarrier = HeaderCarrier()
+
+    "gets a stream when the file exists" in forAll(Gen.stringOfN(20, Gen.alphaNumChar)) {
```

Comment:
Its better to use ``` alphaNumeric ```  Gen already available inside ``` BaseGenerators ```

Reply:
Why is it better to use that, rather than an inbuilt?

Reply:
if there is not benefit then what is the purpose of creating below one inside BaseGenerators?
  def alphaNumeric(length: Int): Gen[String] =
    for {
      listOfChar <- listOfN(length, alphaNumChar)
    } yield listOfChar.mkString

Reply:
Because someone didn't realise that they could do what I did.

> and also I think keeping generator at one place make more sense.

We should stick with the base library one then, and not duplicate it with our own.